File: frontend/src/types/image.ts
export interface ImageMetadata {
    id: string;
    filename: string;
    filepath: string;
    file_size: number;
    bands?: number;
    /** Number of raster bands (same as bands, exposed by new ImageResponse) */
    num_channels?: number;
    width?: number;
    height?: number;
    created_at?: string;
    uploaded_at?: string;
    project_id?: string;
    /**
     * GeoTIFF spatial metadata extracted by rasterio on upload.
     * bounds: [west, south, east, north] in WGS84
     */
    image_metadata?: {
        bounds?: [number, number, number, number] | null;
        crs?: string | null;
        dtype?: string | null;
    } | null;
}

export interface UploadProgress {
    loaded: number;
    total: number;
    percentage: number;
}

File: backend/app/schemas/image.py
from pydantic import BaseModel
from typing import Optional, Any, Dict
from datetime import datetime


class ImageBase(BaseModel):
    filename: str
    project_id: str


class ImageCreate(ImageBase):
    filepath: str
    file_size: int
    width: Optional[int] = None
    height: Optional[int] = None
    num_channels: Optional[int] = None


class ImageResponse(ImageBase):
    id: str
    filepath: str
    file_size: int
    width: Optional[int] = None
    height: Optional[int] = None
    num_channels: Optional[int] = None
    image_metadata: Optional[Dict[str, Any]] = None
    uploaded_at: datetime

    class Config:
        from_attributes = True
